src/lean_spec/subspecs/containers/block/block.py

```diff
@@ -177,24 +180,25 @@ def verify_signatures(
         ):
             validator_ids = aggregated_attestation.aggregation_bits.to_validator_indices()
 
-            assert len(aggregated_signature) == len(validator_ids), (
-                "Aggregated attestation signature count mismatch"
-            )
-
             attestation_root = aggregated_attestation.data.data_root_bytes()
 
-            # Verify each validator's attestation signature
-            for validator_id, signature in zip(validator_ids, aggregated_signature, strict=True):
+            # Verify the leanVM aggregated proof for this attestation
+            for validator_id in validator_ids:
                 # Ensure validator exists in the active set
                 assert validator_id < Uint64(len(validators)), "Validator index out of range"
-                validator = validators[validator_id]
-
-                assert signature.verify(
-                    validator.get_pubkey(),
-                    aggregated_attestation.data.slot,
-                    attestation_root,
-                    scheme,
-                ), "Attestation signature verification failed"
+
+            public_keys = [validators[vid].get_pubkey() for vid in validator_ids]
+            try:
+                verify_aggregated_payload(
+                    public_keys=public_keys,
+                    payload=bytes(aggregated_signature),
+                    message=attestation_root,
+                    epoch=aggregated_attestation.data.slot,
+                )
+            except LeanMultisigError as exc:
+                raise AssertionError(
+                    "Attestation aggregated signature verification failed"
+                ) from exc
```

Comment:
Should we just let the original error propagate up? Not sure if we need to wrap it into a more generic AssertionError type and also suppress the original exception object